File: source/GyeongHo/boj/10866.py
import sys
from collections import deque
input = sys.stdin.readline

n = int(input())
q = deque()
for _ in range(n):
    command = input().rstrip()
    if command == 'pop_front':
        if q:
            print(q.popleft())
        else:
            print(-1)
    elif command == 'pop_back':
        if q:
            print(q.pop())
        else:
            print(-1)
    elif command == 'size':
        print(len(q))
    elif command == 'empty':
        if q:
            print(0)
        else:
            print(1)
    elif command == 'front':
        if q:
            print(q[0])
        else:
            print(-1)
    elif command == 'back':
        if q:
            print(q[-1])
        else:
            print(-1)
    else:
        if command.split()[0] == 'push_front':
            q.appendleft(int(command.split()[1]))
        else:
            q.append(int(command.split()[1]))

File: source/GyeongHo/boj/10866.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
	deque<int> d1;

	int n;
	string inst;
	cin >> n;
	while (n--) {
		cin >> inst;
		if (!inst.compare("push_front")) {
			int x;
			cin >> x;
			d1.push_front(x);
		}
		else if (!inst.compare("push_back")) {
			int x;
			cin >> x;
			d1.push_back(x);
		}
		else if (!inst.compare("pop_front")) {
			if (!d1.empty()) {
				cout << d1.front() << "\n";
				d1.pop_front();
			}
			else
				cout << -1 << "\n";
		}
		else if (!inst.compare("pop_back")) {
			if(!d1.empty()) {
				cout << d1.back() << "\n";
				d1.pop_back();
			}
			else
				cout << -1 << "\n";
		}
		else if (!inst.compare("size")) {
			cout << d1.size() << "\n";
		}
		else if (!inst.compare("empty")) {
			cout << d1.empty() << "\n";
		}
		else if (!inst.compare("front")) {
			if (!d1.empty()) {
				cout << d1.front() << "\n";
			}
			else
				cout << -1 << "\n";
		}
		else if (!inst.compare("back")) {
			if (!d1.empty())
				cout << d1.back() << "\n";
			else
				cout << -1 << "\n";
		}
		else
			cout << "wrong instruction" << "\n";
	}
	return 0;
}
